src/hacbs/components/Commits/CommitSidePanelHeader.tsx

```diff
@@ -0,0 +1,118 @@
+  ButtonVariant,
+  DrawerActions,
+  DrawerCloseButton,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+} from '@patternfly/react-core';
+import AngleDownIcon from '@patternfly/react-icons/dist/js/icons/angle-down-icon';
+import AngleUpIcon from '@patternfly/react-icons/dist/js/icons/angle-up-icon';
+import { runStatus } from '../../../shared';
+import { PipelineRunKind } from '../../types';
+import { statuses } from '../../utils/commits-utils';
+
+export interface SortedPLRList {
+  [runStatus.Running]: PipelineRunKind[];
+  [runStatus.Cancelled]: PipelineRunKind[];
+  [runStatus.Failed]: PipelineRunKind[];
+  [runStatus.Pending]: PipelineRunKind[];
+  [runStatus.Succeeded]: PipelineRunKind[];
+}
+interface CommitSidePanelHeaderProps {
+  isExpanded: boolean;
+  drawerRef: React.Ref<HTMLElement>;
+  currentStatus: string;
+  sortedPLRList: SortedPLRList;
+  changeStatus: (status: string) => void;
+  currentIndex: number;
+  nextTask: () => void;
+  previousTask: () => void;
+  onPanelCloseClick: () => void;
+}
+
+const CommitSidePanelHeader: React.FC<CommitSidePanelHeaderProps> = ({
+  isExpanded,
+  drawerRef,
+  currentStatus,
+  sortedPLRList,
+  changeStatus,
+  currentIndex,
+  nextTask,
+  previousTask,
+  onPanelCloseClick,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const onToggle = (isDropdownOpen: boolean) => {
+    setIsOpen(isDropdownOpen);
+  };
+
+  const onFocus = () => {
+    const element = document.getElementById('toggle-basic');
+    element.focus();
+  };
+
+  const onSelect = () => {
+    setIsOpen(false);
+    onFocus();
+  };
+
+  const dropdownItems = statuses.map((status) => (
+    <DropdownItem
+      onClick={() => {
+        changeStatus(status);
+      }}
+      key={status}
+    >
+      {status}
+    </DropdownItem>
+  ));
+
+  return (
+    <>
+      <span tabIndex={isExpanded ? 0 : -1} ref={drawerRef}>
+        <Dropdown
+          onSelect={onSelect}
+          toggle={
+            <DropdownToggle id="toggle-basic" onToggle={onToggle}>
+              {currentStatus}
+            </DropdownToggle>
+          }
+          isOpen={isOpen}
+          dropdownItems={dropdownItems}
+        />
+        {sortedPLRList[currentStatus].length > 0 && (
+          <span className="pf-u-ml-xs pf-u-mr-xs" data-test="plr-count">
+            {`${currentIndex + 1}/${sortedPLRList[currentStatus].length}`}
+          </span>
+        )}
+        <Button
+          variant={ButtonVariant.control}
+          className="pf-u-p-sm"
+          isDisabled={currentIndex >= sortedPLRList[currentStatus].length - 1}
+          onClick={nextTask}
+          data-test="next-plr-button"
+        >
+          <AngleDownIcon />
+        </Button>
+        <Button
+          variant={ButtonVariant.control}
+          className="pf-u-p-sm"
+          isDisabled={currentIndex < 1}
+          onClick={previousTask}
+          data-test="previous-plr-button"
+        >
+          <AngleUpIcon />
+        </Button>
```

Comment:
The spacing on these buttons don't look like the UX doc.
This is something we can also address afterwards.

But it seems like UX wants what looks like input group attachments. I'm not sure exactly though.
@Ranelim could you help identify the PF components you've used in this mock:
![image](https://user-images.githubusercontent.com/14068621/195392174-1c1dcb31-ee8f-406d-959a-5b9895876bbb.png)

Reply:
icon buttons don't have grey outlines. This is what confused me. If we don't need to have those grey outlines then we're good. I thought it was some other component / styling i wasn't aware of.